core/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -520,4 +526,39 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeBoolean(updateAllTypes);
         waitForActiveShards.writeTo(out);
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (!settings.isEmpty()) {
+            builder.startObject(SETTINGS.getPreferredName());
+            settings.toXContent(builder, params);
+            builder.endObject();
+        }
+
+        if (!mappings.isEmpty()) {
+            builder.startObject(MAPPINGS.getPreferredName());
+
+            for (Map.Entry<String, String> entry : mappings.entrySet()) {
+                builder.rawField(entry.getKey(), new BytesArray(entry.getValue()), XContentType.JSON);
```

Comment:
Seems infelicitous that we store the mappings as JSON-serialized `Strings`, and hence have to deserialize & reserialize them when making an HTTP request. It would presumably be nicer if we stored mappings as typed objects, akin to `Alias`es and `Settings`? Would that be a good idea/outside the scope of this PR?